### PR TITLE
Teskit empty assertdiff

### PR DESCRIPTION
**Pull request by MasseGuillaume**
@olafurpg look what I found :p

I should always test edge cases.

### CODE REVIEW

File: scalafix-core/shared/src/main/scala/scalafix/patch/Patch.scala [hunk 1]
```diff
@@ -205,7 +205,7 @@ object Patch {
       case _: LintPatch => hasLintMessage = true
       case _ => onlyLint = false
     }
-    hasLintMessage && onlyLint
+    patch.isEmpty || hasLintMessage && onlyLint
```
Comment (MasseGuillaume) [inline scalafix-core/shared/src/main/scala/scalafix/patch/Patch.scala]: I wrote an empty syntactic rule and I ran the SemanticSuite. It complained it was missing an output. This is because it checks if there are lint messages. It's obviously not the best way to fix this.

File: scalafix-testkit/src/main/scala/scalafix/testkit/CommentAssertion.scala [hunk 2]
```diff
@@ -353,24 +353,41 @@ object AssertDiff {
...
+          .flatMap(_.headOption.map(_.assert))
+          .toList
+
+      val unexpected =
+        matrix.columns
+          .filter(_.forall(_.isWrong))
+          .flatMap(_.headOption.map(_.lintMessage))
+          .toList
+
+      val missmatch =
```
Comment (olafurpg) [inline scalafix-testkit/src/main/scala/scalafix/testkit/CommentAssertion.scala]: mismatch
